### DISCUSSION TITLE
Awesome coffea

### DISCUSSION THREAD

**Discussion by nsmith-** [scikit-hep/coffea#829] [Show and tell]
A place to collect publicly available examples of physics analyses in CMS or beyond using Coffea!

- [boostedhiggs](https://github.com/nsmith-/boostedhiggs), with many derivatives (HIG-21-012, ...)
- [TopCoffea](https://github.com/TopEFT/topcoffea) (TOP-22-006)

Many examples can be found through github search: https://github.com/search?q=%22import+coffea%22+%28language%3A%22Python%22+OR+language%3A%22Jupyter+Notebook%22%29&type=code&l=Jupyter+Notebook

**Comment by alexander-held** [comment]
We have an [Analysis Grand Challenge](https://agc.readthedocs.io/) implementation making use of `coffea` here: [iris-hep/analysis-grand-challenge](https://github.com/iris-hep/analysis-grand-challenge/) (no physics publications as this is a technical exercise).

**Comment by garvitaa** [comment]
B2G analysis searching for ttbar resonances in dileptonic state is available on GitLab: https://gitlab.cern.ch/gagarwal/ttbardileptonic
This repository is a work in progress as of today.

**Comment by DryRun** [comment]
Measurement of B hadron production fraction ratios using the CMS 2018 B parking data set: https://github.com/DryRun/boffea

CMS hadron calorimeter (HCAL) raw data analysis framework for Prompt Feedback Group (PFG): https://github.com/DryRun/hcalanalysis
(Implements a trivial schema and uses processors, but doesn't really use any coffea physics object methods)

**Comment by lgray** [comment]
Higgs to gamma gamma run 3 framework: https://gitlab.cern.ch/HiggsDNA-project/HiggsDNA

**Comment by lgray** [comment]
https://github.com/bu-cms/bucoffea @AndreasAlbert but it hasn't seen any activity since ~2021. I'm not sure how things have evolved.

**Reply by AndreasAlbert** [reply]
@lgray bucoffea was used for at least two analyses:
* [Run-2 CMS monojet / mono-V](https://arxiv.org/abs/2107.13021)
* [Run-2 CMS VBF H(inv)](https://arxiv.org/abs/2201.11585)

I left physics after these analyses, so don't know what happened after. @alpakpinar might know more.

**Reply by alpakpinar** [reply]
@lgray @AndreasAlbert there is more recent activity in my fork of the repository - especially [this branch](https://github.com/alpakpinar/bucoffea/tree/feature/particlenet/bucoffea) is used for a version of the VBF H(inv) analysis mentioned above, using a ParticleNet tagger (under development at the moment & not very active). 

Also I can add that the JetMET trigger group is using this tool to make it one of the standard ways to measure trigger efficiencies from NanoAOD (current effort mostly by @theochatzis and me). This repository can be found [here](https://github.com/cms-jet/jme-cofftea/tree/main).

**Comment by lgray** [comment]
https://github.com/yhaddad/NUCoffea @yhaddad but it's also not really had commits since 2021. Would be good to see where it's gone!

**Reply by yhaddad** [reply]
@lgray, this repo was used for EXO-19-003, and it has been on hold for quite some time, but various forks are still active and used mostly in [DiHiggs](https://github.com/vivannguyen/HHCoffea) and [EXO](https://github.com/SUEPPhysics/SUEPCoffea). We have a new implementation quite similar to [boostedhiggs](https://github.com/nsmith-/boostedhiggs) used in SMP-23-001.

**Reply by lgray** [reply]
@yhaddad for the new implementation if it's available could you link it?

**Comment by valsdav** [comment]
A configuration layer for Coffea analyses:  https://github.com/PocketCoffea/PocketCoffea

- :boom:  Zmumu example:  [AnalysisConfigs github](https://github.com/PocketCoffea/AnalysisConfigs/tree/main/configs/zmumu)
- :wrench: :  ttHbb parton matching studies: [AnalysisConfigs github](https://github.com/PocketCoffea/AnalysisConfigs/tree/main/configs/ttHbb/semileptonic/parton_matching)
- :book: docs  (WIP) : https://pocketcoffea.readthedocs.io/en/latest/index.html

**Comment by lgray** [comment]
https://github.com/GageDeZoort/azh_coffea/tree/main (HIG-22-004) @GageDeZoort - fun uses of numba in this one (fastmtt)

**Comment by btcardwell** [comment]
Analysis framework for in-progress search for Self-Interacting Dark Matter: [https://github.com/btcardwell/SIDM](https://github.com/btcardwell/SIDM)
